### PR TITLE
Add support for Go Revive linter

### PR DESCRIPTION
**Pull request by mennanov**
This PR adds support for a popular Go linter https://github.com/mgechev/revive

The code has a README.md file and an example directory.

I've signed the CLA.

### CODE REVIEW

File: gorevive/deps.bzl [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+load("@bazel_tools//tools/build_defs/repo:git.bzl", "new_git_repository")
+
+
+def go_revive_dependencies():
+    
+    _maybe(
+        go_repository,
+        name = "com_github_burntsushi_toml",
+        importpath = "github.com/BurntSushi/toml",
+        tag = "v0.3.0",
```
Comment (ash2k) [inline gorevive/deps.bzl]: Would be better to specify commit revisions here and below - they are immutable unlike tags.

Comment (mennanov) [reply gorevive/deps.bzl]: These deps are generated by Gazelle from https://github.com/mgechev/revive/blob/master/go.mod
I think it is safer to stick to whatever the author of revive has chosen (tags vs commits).
What do you think?

Comment (ash2k) [reply gorevive/deps.bzl]: Yeah, lets keep them in that case, that's fine.

**Comment by ash2k** [comment]
Thanks!

**Pull request merged**